core/rawdb/database.libevm_test.go

```diff
@@ -0,0 +1,193 @@
+				default:
+					modified = append(modified, line)
+				}
+			}
+			// Add lines for data collected with [rawdb.WithDatabaseStatRecorder]
+			line := []string{"My database", "My category", myStat.Size(), myStat.Count()}
+			modified = append(modified, line)
+			return modified
+		}),
+	}
+
+	err := rawdb.InspectDatabase(db, keyPrefix, keyStart, options...)
+	if err != nil {
+		fmt.Println(err)
+	}
+	// Output:
+	// +-----------------+-------------------------+---------+-------+
+	// |    DATABASE     |        CATEGORY         |  SIZE   | ITEMS |
+	// +-----------------+-------------------------+---------+-------+
+	// | Key-Value store | Headers                 | 0.00 B  |     0 |
+	// | Key-Value store | Bodies                  | 0.00 B  |     0 |
+	// | Key-Value store | Receipt lists           | 0.00 B  |     0 |
+	// | Key-Value store | Block number->hash      | 0.00 B  |     0 |
+	// | Key-Value store | Block hash->number      | 0.00 B  |     0 |
+	// | Key-Value store | Transaction index       | 0.00 B  |     0 |
+	// | Key-Value store | Bloombit index          | 6.00 B  |     1 |
+	// | Key-Value store | Contract codes          | 0.00 B  |     0 |
+	// | Key-Value store | Hash trie nodes         | 0.00 B  |     0 |
+	// | Key-Value store | Path trie state lookups | 0.00 B  |     0 |
+	// | Key-Value store | Path trie account nodes | 0.00 B  |     0 |
+	// | Key-Value store | Path trie storage nodes | 0.00 B  |     0 |
+	// | Key-Value store | Trie preimages          | 0.00 B  |     0 |
+	// | Key-Value store | Account snapshot        | 0.00 B  |     0 |
+	// | Key-Value store | Storage snapshot        | 0.00 B  |     0 |
+	// | Key-Value store | Beacon sync headers     | 0.00 B  |     0 |
+	// | Key-Value store | Clique snapshots        | 0.00 B  |     0 |
+	// | Key-Value store | Singleton metadata      | 20.00 B |     1 |
+	// | Light client    | CHT trie nodes          | 0.00 B  |     0 |
+	// | Light client    | Bloom trie nodes        | 0.00 B  |     0 |
+	// | My database     | My category             | 12.00 B |     1 |
+	// +-----------------+-------------------------+---------+-------+
+	// |                            TOTAL          | 38.00 B |       |
+	// +-----------------+-------------------------+---------+-------+
+}
+
+type stubDatabase struct {
+	ethdb.Database
+	iterator ethdb.Iterator
+}
+
+func (s *stubDatabase) NewIterator(keyPrefix, keyStart []byte) ethdb.Iterator {
+	return s.iterator
+}
+
+// AncientSize is used in [InspectDatabase] to determine the ancient sizes.
+func (s *stubDatabase) AncientSize(kind string) (uint64, error) {
+	return 0, nil
+}
+
+func (s *stubDatabase) Ancients() (uint64, error) {
+	return 0, nil
+}
+
+func (s *stubDatabase) Tail() (uint64, error) {
+	return 0, nil
+}
+
+func (s *stubDatabase) Get(key []byte) ([]byte, error) {
+	return nil, nil
+}
+
+func (s *stubDatabase) ReadAncients(fn func(ethdb.AncientReaderOp) error) (err error) {
```

Comment:
nit unneeded named return

```suggestion
func (s *stubDatabase) ReadAncients(fn func(ethdb.AncientReaderOp) error) error {
```

Reply:
You can't merge a PR into my branch only to nit-pick your own code 🤣 haha, do that first 😛